src/components/views/rooms/wysiwyg_composer/hooks/useWysiwygEditActionHandler.ts

```diff
@@ -39,9 +48,17 @@ export function useWysiwygEditActionHandler(disabled: boolean, composerElement:
                 case Action.FocusEditMessageComposer:
                     focusComposer(composerElement, context, roomContext, timeoutId);
                     break;
+                case Action.ComposerInsert:
+                    if (payload.timelineRenderingType !== roomContext.timelineRenderingType) break;
+                    if (payload.composerType !== ComposerType.Edit) break;
+
+                    if (payload.text) {
+                        setSelection(composerContext.selection).then(() => composerFunctions.insertText(payload.text));
+                    }
+                    break;
```

Comment:
`ComposerInsert` is [documented](https://github.com/matrix-org/matrix-react-sdk/blob/06aa2af560d8f33432058830b1744170465a72e0/src/dispatcher/actions.ts#L205) as:

>  Inserts content into the active composer. Should be used with ComposerInsertPayload.

It doesn't mention anything about changing the selection or checking the timeline rendering type. Maybe it should?

Reply:
In the previous composer, the selection change is made in the EditorMode, in the call of insertText. 

Why ? Because when we are clicking or using the modal, the composer loses the focus and its selection. The selection is saved on blur and set when needed (here for example)

Reply:
that's fine. Just saying, maybe the documentation for `ComposerInsert` should be updated to reflect the current behaviour, even if that means noting the different behaviour for the two composers.